### ISSUE TITLE
Brave Extension high memory usage with WDP enabled

### ISSUE THREAD

**Issue by atuchin-m** [brave/brave-browser#23573]
## Description 
A memory leak with enabled WebDiscovery Project in Brave Extension.
Community thread: https://community.brave.com/t/memory-leak-with-no-extensions/405082

## Steps to Reproduce  (need to be clarified)
   1.  Enable Web Discovery project in brave://settings;
   2.  Restart the browser;
   3.  Open ~10 youtube tabs, scrolls them down;
   4.  (Optional) watch videos for some time, navigate on youtube.com inside the tabs;
   5. Check Brave Extension memory usage.

## Actual result:
Brave extension memory footprint < 100MB

## Expected result:
Brave extension memory footprint > 200MB

Extra details:
1. No issue without enabled WDP;
2. Reloading the background page helps for some time => memory allocation is triggered by the background page;
3. JS memory is ~10-20Mb and most of the memory is taken by DOM elements. For the extension background page use webtools => memory => heap snapshot to inspect it.

![image (17)](https://user-images.githubusercontent.com/45488748/174831043-101aeeed-acf9-4931-8824-98f89d49e7df.png)
![image (18)](https://user-images.githubusercontent.com/45488748/174831088-7842a98d-55b3-49e6-b74a-751bab5641f8.png)
![image (19)](https://user-images.githubusercontent.com/45488748/174831200-8da513b4-3a95-4da4-bee5-bcce2f2f75ce.png)

**Comment by stephendonner** [comment]
## Verified `PASSED` using 

Brave | 1.43.61 Chromium: 104.0.5112.81 (Official Build) beta (x86_64)
-- | --
Revision | 5b7b76419d50f583022568b6764b630f6ddc9208-refs/branch-heads/5112@{#1309}
OS | macOS Version 11.6.8 (Build 20G730)

First, reproduced the issue with `1.42.88`:

10 YouTube tabs | 0 open tabs (after 5+ min)
-----------------|---------------------------
<img width="570" alt="Screen Shot 2022-08-08 at 12 16 17 PM" src="https://user-images.githubusercontent.com/387249/183500346-e6201849-e873-4bba-bc8f-8b9d5750a058.png"> | <img width="570" alt="Screen Shot 2022-08-08 at 12 21 51 PM" src="https://user-images.githubusercontent.com/387249/183500396-c1ab6588-3dc6-4ab4-a492-f1e771363428.png">

Confirmed fixed using `1.43.61`.

### Steps:
1. install `1.43.61`
2. launch Brave
3. open `brave://settings/search` and toggle `Web Discovery Project` to `On`/`Enabled`
4. load 10 YouTube tabs with high bitrates
5. close all tabs (the window), but don't shut down Brave
6. waited 5-10 minutes

#### Confirmed memory allocation/use was down from `225` -> `66.1 MB`

10 YouTube tabs | 0 open tabs (after 5+ min)
-----------------|---------------------------
<img width="570" alt="Screen Shot 2022-08-08 at 12 30 16 PM" src="https://user-images.githubusercontent.com/387249/183500644-ad2d31ec-70ec-4b69-a83d-4afe32287dac.png"> | <img width="570" alt="Screen Shot 2022-08-08 at 12 36 10 PM" src="https://user-images.githubusercontent.com/387249/183500650-467160c0-595e-4df5-97e4-95dbeb1fcfae.png">

**Comment by MadhaviSeelam** [comment]
Verification `In progress` with 
```
Brave | 1.43.67 Chromium: 104.0.5112.81 (Official Build) beta (64-bit)
-- | --
Revision | 5b7b76419d50f583022568b6764b630f6ddc9208-refs/branch-heads/5112@{#1309}
OS | Windows 11 Version 21H2 (Build 22000.856)
```

Reproduced with `1.42.88`

10 YouTube tabs open|0 open tabs(after 5+ min)
----------|--------
<image width="950" alt="image" src="https://user-images.githubusercontent.com/98358127/184199915-2c228be8-8013-4afa-b7c8-b64a511505cc.png">|<image width="950" alt="image" src="https://user-images.githubusercontent.com/98358127/184207693-8d4b84fc-d9c7-45df-ba91-c92205fff229.png">

After Fix
---------
### Steps:
1. install `1.43.67`
2. launch Brave
3. open `brave://settings/search` and toggle `Web Discovery Project` to `On`/`Enabled`
4. load 10 YouTube tabs 
5. close all tabs (the window), but don't shut down Brave
6. waited 5-10 minutes

#### Confirmed memory allocation/use was down `106` MB`

Step 3|Step 4|Step 5
----|----|-----
<image width="950" alt="image" src="https://user-images.githubusercontent.com/98358127/184192856-6478d6ee-6dde-4d1b-b724-8bc5578aa71a.png">|<image width="950" alt="image" src="https://user-images.githubusercontent.com/98358127/184196493-50d4217c-9bc7-470a-bcb1-31e51481618f.png">|<image width="950" alt="image" src="https://user-images.githubusercontent.com/98358127/184198728-42ad8a9a-ff99-4b09-ad84-35616104eb9c.png">

**Comment by btlechowski** [comment]
Verification passed on

Brave | 1.43.78 Chromium: 104.0.5112.102 (Official Build) beta (64-bit)
-- | --
Revision | 8e5396254975ef939f2ef7d0bd334e48a052b536-refs/branch-heads/5112@{#1478}
OS | Ubuntu 18.04 LTS

First, reproduced the issue with `1.42.x`:

![image](https://user-images.githubusercontent.com/34715963/185980220-83f0be4b-b631-426b-b564-74042ace2c53.png)


Confirmed fixed using `1.43.x`.

![image](https://user-images.githubusercontent.com/34715963/185976801-dcfa046b-3b3c-48f7-bbc6-f8b80f822204.png)
